crates/uv-workspace/src/pyproject_mut.rs

```diff
@@ -520,7 +519,16 @@ pub fn add_dependency(
         }
         // Cannot perform ambiguous updates.
         _ => Err(Error::Ambiguous),
-    }
+    };
+
+    deps.sort_by_key(|d| {
```

Comment:
seemed like this was going to be the cleanest way to get this done. while it can be wasteful (we may end up sorting every time we add a package if `uv add` invocation adds multiple packages?), i imagine most invocations of `uv add` will:
1. add a few dependencies manually
2. sorting is not expensive with the number of dependencies being sorted here
(probably?)

Reply:
actually realized this probably isn't the correct fix since we probably should preserve the sorted order on removal as well.

will fix (maybe this weekend)

Reply:
We should only do this when the dependencies are already sorted, if the user has a custom sorting we shouldn't change that and just add at the end. So the idea would be that we first check if the list is sorted, and if it is, scan for the insertion point.

Reply:
ahh good call.